src/process.ml

```diff
@@ -92,6 +92,59 @@ module Output = struct
     { t with status = Keep_open }
 end
 
+module Input = struct
```

Comment:
I'm not convinced all this duplication is necessary. Would it be possible to repurpose the output module instead? Call it something more generic and make the channel the type variable. We could even use a GADT to make sure that in_channel and out_channel are the only possibilities but that isn't essential.

Reply:
The last commit implements this (GADTs included :)).